osu.Game/Modes/Mod.cs

```diff
@@ -0,0 +1,60 @@
+﻿//Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
+//Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System.ComponentModel;
+using osu.Game.Graphics;
+
+namespace osu.Game.Modes
+{
+    public class Mod
+    {
+        public ModName Name;
+        public ModType Type;
+
+        public FontAwesome Icon;
+
+        public double ScoreMultiplier;
+    }
+
+    public enum ModType
+    {
+        DiffReducing,
+        DiffIncreasing,
+        Assistance
+    }
+    public enum ModName
+    {
+        [Description(@"Autopilot")]

```

Comment:
not sure i agree with acronyms here.

probably best to use the mods enum for stable for now (feel free to add description attributes in!)

```
[Flags]
    public enum Mods
    {
        None                   = 0,
        NoFail                 = 1 << 0,
        Easy                   = 1 << 1,
        //NoVideo              = 1 << 2,
        Hidden                 = 1 << 3,
        HardRock               = 1 << 4,
        SuddenDeath            = 1 << 5,
        DoubleTime             = 1 << 6,
        Relax                  = 1 << 7,
        HalfTime               = 1 << 8,
        Nightcore              = 1 << 9,
        Flashlight             = 1 << 10,
        Autoplay               = 1 << 11,
        SpunOut                = 1 << 12,
        Relax2                 = 1 << 13,
        Perfect                = 1 << 14,
        Key4                   = 1 << 15,
        Key5                   = 1 << 16,
        Key6                   = 1 << 17,
        Key7                   = 1 << 18,
        Key8                   = 1 << 19,
        FadeIn                 = 1 << 20,
        Random                 = 1 << 21,
        Cinema                 = 1 << 22,
        Target                 = 1 << 23,
        Key9                   = 1 << 24,
        KeyCoop                = 1 << 25,
        Key1                   = 1 << 26,
        Key3                   = 1 << 27,
        Key2                   = 1 << 28,
        LastMod                = 1 << 29,
        KeyMod = Key1 | Key2 | Key3 | Key4 | Key5 | Key6 | Key7 | Key8 | Key9 | KeyCoop,
        FreeModAllowed = NoFail | Easy | Hidden | HardRock | SuddenDeath | Flashlight | FadeIn | Relax | Relax2 | SpunOut | KeyMod,
        ScoreIncreaseMods = Hidden | HardRock | DoubleTime | Flashlight | FadeIn
    }
```